File: scraper/scraper-api/parse.js
export default function parseMessage(rawText) {
  const lines = rawText.split("\n").map(l => l.trim());

  return {
    name: extractValue(lines, "応募者名"),
    loginId: extractValue(lines, "メールアドレス"),
    password: extractValue(lines, "パスワード")
  }
}

function extractValue(lines, keyword) {
  for (const line of lines) {
    if (line.includes(keyword)) {
      // 「:」または「：」で分割
      const parts = line.split(/[:]|[：]/);
      if (parts.length > 1) {
        // 「様」がある場合は削除
        return parts[1].trim().replace("様", "");
      }
    }
  }
  return null;
}

File: scraper/scraper-api/python/parsers/parse.py
import re

def parse_message(message: str) -> dict:
    # メールの本文からログインIDやパスワード、応募者名を抽出
    lines = message.split("\n")

    def extract(keywords):
        for line in lines:
            for kw in keywords:
                if kw in line:
                    parts = re.split("：|:", line)
                    if len(parts) > 1:
                        return parts[1].strip().replace("様", "")
        return None

    return {
        "loginId": extract(["メールアドレス"]),
        "password": extract(["パスワード"]),
        "name": extract(["応募者名"])
    }
